File: src/main/java/pl/tsimur/jpatest/model/Role.java
package pl.tsimur.jpatest.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

import static javax.persistence.EnumType.STRING;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table
public class Role {
    @Id
    @Enumerated(STRING)
    private Scope role;

    @ManyToMany(mappedBy = "roles")
    @JsonIgnore
    private List<User> users = new ArrayList<>();
}

File: src/main/kotlin/pl/timur/jpatest/model/Role.kt
package pl.timur.jpatest.model

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*

@Entity
@Table
data class Role(
    @Id
    @Enumerated(EnumType.STRING)
    val role: Scope
) {
    @ManyToMany(mappedBy = "roles")
    @JsonIgnore
    val users: List<User> = ArrayList()
}
